torchx/schedulers/local_scheduler.py

```diff
@@ -168,6 +168,7 @@ class LocalOpts(TypedDict, total=False):
     log_dir: str
     prepend_cwd: Optional[bool]
     auto_set_cuda_visible_devices: Optional[bool]
+    auto_set_cuda_visible_devices_ids: List[str]
```

Comment:
Should the list be an optional? I think we need an explicit optional semantics instead of empty list.

Reply:
TorchX right now does not understand optional scheduler arguments. 

```
Cache hits: 100%. Commands: 85 (cached: 85, remote: 0, local: 0)
Traceback (most recent call last):
  File "<string>", line 51, in <module>
  File "<string>", line 37, in __run
  File "/usr/local/fbcode/platform010/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/fbcode/platform010/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/cli/main.py", line 120, in <module>
    main()
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/cli/main.py", line 116, in main
    run_main(get_sub_cmds(), argv)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/cli/main.py", line 112, in run_main
    args.func(args)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/cli/cmd_run.py", line 248, in run
    self._run(runner, args)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/cli/cmd_run.py", line 184, in _run
    cfg = scheduler_opts.cfg_from_str(args.scheduler_args)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/specs/api.py", line 866, in cfg_from_str
    cfg[key] = _cast_to_type(val, runopt_.opt_type)
  File "/data/users/jmiquel/fbsource/buck-out/v2/gen/fbcode/da4de3c780a17bfa/torchx/cli/__torchx__/torchx#link-tree/torchx/specs/api.py", line 860, in _cast_to_type
    return opt_type(value)
  File "/usr/local/fbcode/platform010/lib/python3.10/typing.py", line 957, in __call__
    result = self.__origin__(*args, **kwargs)
  File "/usr/local/fbcode/platform010/lib/python3.10/typing.py", line 387, in __call__
    raise TypeError(f"Cannot instantiate {self!r}")
TypeError: Cannot instantiate typing.Union
```

I will look at whether it can be cleanly added to the type instantiation.